src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_auriaya.cpp

```diff
@@ -63,329 +62,233 @@ enum
     NPC_SEEPING_FERAL_ESSENCE           = 34098,            // summoned by the feral defender on feign death
     // NPC_GUARDIAN_SWARN               = 34034,            // summoned by spell
     NPC_FERAL_DEFENDER_STALKER          = 34096,
+
+    SOUND_DEATH                         = 15476,
 };
 
 /*######
 ## boss_auriaya
 ######*/
 
-struct boss_auriayaAI : public ScriptedAI
+enum AuriayaActions
 {
-    boss_auriayaAI(Creature* pCreature) : ScriptedAI(pCreature)
-    {
-        m_pInstance = (instance_ulduar*)pCreature->GetInstanceData();
-        m_bIsRegularMode = pCreature->GetMap()->IsRegularDifficulty();
-        Reset();
-    }
-
-    instance_ulduar* m_pInstance;
-    bool m_bIsRegularMode;
-
-    uint32 m_uiEnrageTimer;
-    uint32 m_uiSwarmTimer;
-    uint32 m_uiSonicScreechTimer;
-    uint32 m_uiSentinelBlastTimer;
-    uint32 m_uiTerrifyingScreechTimer;
-    uint32 m_uiDefenderTimer;
-
-    void Reset() override
-    {
-        m_uiEnrageTimer             = 10 * MINUTE * IN_MILLISECONDS;
-        m_uiSwarmTimer              = 50000;
-        m_uiSonicScreechTimer       = 58000;
-        m_uiSentinelBlastTimer      = 40000;
-        m_uiTerrifyingScreechTimer  = 38000;
-        m_uiDefenderTimer           = 1 * MINUTE * IN_MILLISECONDS;
-    }
-
-    void JustDied(Unit* /*pKiller*/) override
-    {
-        DoScriptText(SAY_DEATH, m_creature);
-
-        if (m_pInstance)
-            m_pInstance->SetData(TYPE_AURIAYA, DONE);
-    }
+    AURIAYA_DEFENDER,
+    AURIAYA_ACTIONS_MAX,
+};
 
-    void Aggro(Unit* /*pWho*/) override
+struct boss_auriayaAI : public BossAI
+{
+    boss_auriayaAI(Creature* creature) : BossAI(creature, AURIAYA_ACTIONS_MAX),
+        m_instance(dynamic_cast<instance_ulduar*>(creature->GetInstanceData())),
```

Comment:
Static cast is enough afaik.

Reply:
It usually is, but with this I can be 100% certain that the instance is what I think it will be. Yes dynamic_cast is more expensive than static_cast, but in this case it's only invoked once at construction, so I believe the overall impact should be negligible in return for guaranteeing that I can't accidentally call some `m_instance->` function that is only available in `instance_ulduar` if someone spawns this Boss outside of Ulduar for a giggle for example